Microsoft.Azure.Cosmos/src/Telemetry/VmMetadataApiHandler.cs

```
@@ -34,28 +34,43 @@ internal static class VmMetadataApiHandler
 
         private static bool isInitialized = false;
         private static AzureVMMetadata azMetadata = null;
-       
         internal static void TryInitialize(CosmosHttpClient httpClient)
         {
-            if (VmMetadataApiHandler.isInitialized)
+            bool isVMMetadataAccessDisabled = false; // Default value
+            try
             {
-                return;
+                isVMMetadataAccessDisabled = ConfigurationManager.GetEnvironmentVariable<bool>("COSMOS_DISABLE_VM_METADATA_ACCESS", false);
```

Looking at other places where we call GetEnvironmentVariable i don't see any special-casing of exceptions - is this try/catch really needed? What exception would we possibly expect?

---

My thought behind that was, if somebody is messing with this env variable (by passing invalid value
), then just fall back to default behavior.
But drawback is, it might confuse some people. so on second thought, removing this try-catch and let it throw the env related exception. Client initialization won't be possible with invalid of this env variable.